### PR TITLE
Completed test, refactored some things, changed method to return a bo…

### PR DESCRIPTION
**Pull request by daneduijnkerke**
…olean on proof verification, fixed createProofFromCredential method to make sure it can create proofs when attributes need to come from a credential in stead of self-attested.

### CODE REVIEW

File: src/main/java/nl/quintor/studybits/indy/wrapper/Verifier.java [hunk 1]
```diff
@@ -26,11 +27,9 @@ public Verifier(IndyWallet wallet) {
...
 
     public CompletableFuture<List<ProofAttribute>> getVerifiedProofAttributes(ProofRequest proofRequest, Proof proof) {
-        Map<String, CredentialIdentifier> identifierMap = proof.getIdentifiers()
-                .stream()
-                .collect(Collectors.toMap(CredentialIdentifier::getCredDefId, Function.identity()));
-        return getEntitiesFromLedger(identifierMap)
-                .thenCompose(wrapException(entitiesFromLedger -> validateProof(proofRequest, proof, entitiesFromLedger)))
+
+        return  validateProof(proofRequest, proof)
+                .thenAccept(result -> ValidateResult(result, "Invalid proof: verifierVerifyProof failed."))
```
Comment (pimotte) [inline src/main/java/nl/quintor/studybits/indy/wrapper/Verifier.java]: Could you change this function name to match Java style (Yes, I'm aware it was already there :sweat_smile: )

File: src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java [hunk 2]
```diff
@@ -124,16 +155,21 @@ public void fullScenarioTest() throws Exception {
         selfAttestedAttributes.put("last_name", "Garcia");
         selfAttestedAttributes.put("phone_number", "123phonenumber");
 
+        //Alice receives proof request from ACME
         ProofRequest decryptedProofRequest = MessageEnvelope.<ProofRequest>parseFromString(jobApplicationProofRequestEnvelope, alice).getMessage();
-
+        //Alice fufills proof request
+        //Create proof object from available credenties for this proof request and self-attested attributes
```
Comment (pimotte) [inline src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java]: Typo "credenties"

File: src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java [hunk 3]
```diff
@@ -142,23 +178,177 @@ public void fullScenarioTest() throws Exception {
...
                 ));
+
+        //ACME Validates proof
+        Boolean isValidProof = new Verifier(acmeWallet).validateProof(jobApplicationProofRequest, decryptedProof).get();
+        System.out.println("Degree validation: " + isValidProof);
+
+        Assert.assertTrue(isValidProof);
+
+        //ACME creates a new credential offer (Job certificate) for Alice
+        if(isValidProof) {
```
Comment (pimotte) [inline src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java]: This if seems unnecessary (if it is necessary, the else should probably assert something?)

File: src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java [hunk 4]
```diff
@@ -142,23 +178,177 @@ public void fullScenarioTest() throws Exception {
                 new ProofAttribute("attr5_referent", "ssn", "123-45-6789"),
                 new ProofAttribute("attr6_referent", "phone_number", "123phonenumber")
                 ));
+
+        //ACME Validates proof
+        Boolean isValidProof = new Verifier(acmeWallet).validateProof(jobApplicationProofRequest, decryptedProof).get();
```
Comment (pimotte) [inline src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java]: I'd make this an unboxed boolean for consistency

File: src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java [hunk 5]
```diff
@@ -142,23 +178,177 @@ public void fullScenarioTest() throws Exception {
...
+
+        }
+
+        //Thrift Sends proof request to Alice
+        String loanApplicationPersonalDetailsProofRequestEnvelope = MessageEnvelope.fromAuthcryptable(loanApplicationPersonalDetailsProofRequest, IndyMessageTypes.PROOF_REQUEST, thrift).toJSON();
+
+
+        //Alice receives proof request from Thrift
+        decryptedProofRequest = MessageEnvelope.<ProofRequest>parseFromString(loanApplicationPersonalDetailsProofRequestEnvelope, alice).getMessage();
+        System.out.println("00000000000000000000000000000000000000: " + decryptedProofRequest);
```
Comment (pimotte) [inline src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java]: Remaining debug logging

File: src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java [hunk 6]
```diff
@@ -142,23 +178,177 @@ public void fullScenarioTest() throws Exception {
...
+        //#Step 4.1.9
+        //Faber creates a connection response with its created DID and Nonce from the received request from Steward
         String newcomerConnectionResponseString =  MessageEnvelope.fromAnoncryptable(newcomerConnectionResponse, IndyMessageTypes.CONNECTION_RESPONSE, newcomer).toJSON();
+
+        //#Step 4.1.13
+        //Steward decrypts the anonymously encrypted message from Faber
         ConnectionResponse connectionResponse = MessageEnvelope.<ConnectionResponse>parseFromString(newcomerConnectionResponseString, steward).getMessage();
+
+        //#Step 4.1.14#? & 4.1.15
+        //Steward authenticates Faber #?
```
Comment (pimotte) [inline src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java]: Remove "#?" here and line above

File: src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java [hunk 7]
```diff
@@ -23,56 +21,85 @@
 
     @Test
     public void fullScenarioTest() throws Exception {
+        //Clear indy_client directory
         removeIndyClientDirectory();
+
+        //#Step 2
```
Comment (pimotte) [inline src/test/java/nl/quintor/studybits/indy/wrapper/MessageScenarioIT.java]: Can you add a link to the page where the numbers refer to?

File: src/main/java/nl/quintor/studybits/indy/wrapper/util/PoolUtils.java [hunk 8]
```diff
@@ -21,12 +21,19 @@ private static File createGenesisTxnFile(String filename, String testPoolIp) thr
...
+                String.format("{\"data\":{\"alias\":\"Node4\",\"blskey\":\"2zN3bHM1m4rLz54MJHYSwvqzPchYp8jkHswveCLAEJVcX6Mm1wHQD1SkPYMzUDTZvWvhuE6VNAkK3KxVeEmsanSmvjVkReDeBEMxeDaayjcZjFGPydyey1qxBHmTvAnBKoPydvuTAqx5f7YNNRAdeLmUi99gERUU7TD8KfAa6MpQ9bw\",\"client_ip\":\"%s\",\"client_port\":9708,\"node_ip\":\"%s\",\"node_port\":9707,\"services\":[\"VALIDATOR\"]},\"dest\":\"4PS3EDQ3dW1tci1Bp6543CfuuebjFrg36kLAUcskGfaA\",\"identifier\":\"TWwCRQRZ2ZHMJFn9TzLp7W\",\"txnId\":\"aa5e817d7cc626170eca175822029339a444eb0ee8f0bd20d3b0b76e566fb008\",\"type\":\"0\"}", testPoolIp, testPoolIp)
+        };
+//		PROTOCOL VERSION 2
```
Comment (pimotte) [inline src/main/java/nl/quintor/studybits/indy/wrapper/util/PoolUtils.java:31]: Please add a TODO marker here "switch to v2"

**Review by pimotte — CHANGES_REQUESTED** [review]
Looks good in general. Couple of small comments. 

Can you also run auto-format with default intellij settings? Especially the lack of spacing between '//' and the comment itself creates some inconsistency.

**Comment by daneduijnkerke** [comment]
Updated PR based on your feedback

**Review by pimotte — APPROVED** [review]

**Pull request merged**